### ISSUE TITLE
Bug: Not seeing the token refresh process happening

### ISSUE THREAD

**Issue by asierraq** [soofstad/react-oauth2-pkce#178]
The auth server I'm using doesn't return JWT tokens so I'm using the following settings to deal with the token refresh process:

```javascript
const authConfig: TAuthConfig = {
  ...
  // Example to redirect back to original path after login has completed
  preLogin: () => localStorage.setItem('preLoginPath', window.location.pathname),
  postLogin: () => window.location.replace(localStorage.getItem('preLoginPath') || ''),
  tokenExpiresIn: 120,
  refreshTokenExpiresIn: 259200,
  onRefreshTokenExpire: (event) => event.logIn(undefined, undefined)
  ...
}
```

I was expecting for the library to do the token refresh process on the background after `tokenExpiresIn` seconds, but what happens is that after `tokenExpiresIn` seconds the user is redirected to the auth server authorization page. Is this expected behavior? Am I misunderstanding something?

## Steps To Reproduce

1. Set `tokenExpiresIn` in the settings with a low value, and a `refreshTokenExpiresIn` with a higher value.
2. Wait `tokenExpiresIn` seconds.

## The current behavior

The user is redirected to the auth server authorization page

## The expected behavior

I was expecting for the library to do the token refresh process on the background after `tokenExpiresIn` seconds

**Comment by soofstad** [comment]
Hi, thank you for reporting.

I will try to recreate the bug when I get some time, but in the mean time, could you post the network log, and state of localStorage? The behavior you're describing will happen if there is no refreshToken, or the refreshRequest fails. Are you sure you have a valid refreshToken?

**Comment by asierraq** [comment]
Nevermind, I was able to debug the auth service to check that the library is calling the `token` endpoint for refreshing the token and it's indeed making the request, but the server was expecting the scope I used when making the authorize request to be the same in the `token` refresh request, and since I was not setting the `scope` in the settings, the authorize request didn't have a scope and the auth server was assuming the scope was all of the scopes allowed. Then when I did the `token` refresh request the library send the same scope (an empty scope) and since this is different to the scope assumed by the auth server (all of the scopes) it was returning a 400. I set the scope setting and the token refresh process is working now. 

Sorry for the noise
